File: src/main/java/um/haberes/core/hexagonal/contabilidad/cargo_clase_imputacion/domain/model/CargoClaseImputacion.java
package um.haberes.core.hexagonal.contabilidad.cargo_clase_imputacion.domain.model;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CargoClaseImputacion {

    private Long cargoClaseImputacionId;

    private Integer dependenciaId;

    private Integer facultadId;

    private Integer geograficaId;

    private Long cargoClaseId;

    private BigDecimal cuentaSueldos;

    private BigDecimal cuentaAportes;

    public String key() {
        return dependenciaId + "." + facultadId + "." + geograficaId + "." + cargoClaseId;
    }
}

File: src/main/java/um/haberes/core/kotlin/model/CargoClaseImputacion.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["dependenciaId", "facultadId", "geograficaId", "cargoClaseId"])])
data class CargoClaseImputacion(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cargoClaseImputacionId: Long? = null,

    var dependenciaId: Int? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var cargoClaseId: Long? = null,
    var cuentaSueldos: BigDecimal? = null,
    var cuentaAportes: BigDecimal? = null,

    ) : Auditable() {
    fun key(): String {
        return dependenciaId.toString() + "." + facultadId + "." + geograficaId + "." + cargoClaseId
    }

}
